File: backend/app/api/insights.py
from fastapi import APIRouter
from app.models.schema import (
    InsightStatsResponse,
    TrendResponse,
    DistributionResponse,
    ComparisonResponse,
    ActivitiesResponse,
)

router = APIRouter()

@router.get("/insights/stats", response_model=InsightStatsResponse)
async def get_insight_stats(track: str, period: str = "30d"):
    """获取统计卡片数据"""
    from app.services.processor import processor
    return processor.get_insight_stats(track, period)

@router.get("/insights/trends", response_model=TrendResponse)
async def get_insight_trends(track: str, period: str = "30d"):
    """获取内容趋势数据"""
    from app.services.processor import processor
    return processor.get_insight_trends(track, period)

@router.get("/insights/distribution", response_model=DistributionResponse)
async def get_insight_distribution(track: str, period: str = "30d"):
    """获取内容分布数据"""
    from app.services.processor import processor
    return processor.get_insight_distribution(track, period)

@router.get("/insights/comparison", response_model=ComparisonResponse)
async def get_insight_comparison():
    """获取赛道对比数据"""
    from app.services.processor import processor
    return processor.get_insight_comparison()

@router.get("/insights/activities", response_model=ActivitiesResponse)
async def get_insight_activities(track: str, limit: int = 5):
    """获取最新动态"""
    from app.services.processor import processor
    return processor.get_insight_activities(track, limit)

File: frontend/src/services/insights.ts
const API_BASE = '/api'

export interface InsightStats {
  content_count: number
  content_count_change: number
  source_count: number
  source_count_change: number
  sentiment_index: number
  sentiment_change: number
  heat_index: number
  heat_rank: number
}

export interface InsightStatsResponse {
  success: boolean
  data?: InsightStats
}

export interface DailyTrend {
  date: string
  positive: number
  negative: number
}

export interface TrendData {
  period: string
  trends: DailyTrend[]
}

export interface TrendResponse {
  success: boolean
  data?: TrendData
}

export interface CategoryDistribution {
  name: string
  color: string
  percentage: number
  count: number
}

export interface DistributionData {
  positive_rate: number
  categories: CategoryDistribution[]
}

export interface DistributionResponse {
  success: boolean
  data?: DistributionData
}

export interface TrackComparison {
  id: string
  name: string
  color: string
  content_count: number
  heat_index: number
  trend: number
}

export interface ComparisonData {
  tracks: TrackComparison[]
}

export interface ComparisonResponse {
  success: boolean
  data?: ComparisonData
}

export interface ActivityItem {
  id: string
  type: string
  type_name: string
  title: string
  source: string
  time_ago: string
  url: string
}

export interface ActivitiesData {
  activities: ActivityItem[]
}

export interface ActivitiesResponse {
  success: boolean
  data?: ActivitiesData
}

export async function getInsightStats(track: string, period: string): Promise<InsightStatsResponse> {
  const response = await fetch(`${API_BASE}/insights/stats?track=${encodeURIComponent(track)}&period=${encodeURIComponent(period)}`)
  return response.json()
}

export async function getInsightTrends(track: string, period: string): Promise<TrendResponse> {
  const response = await fetch(`${API_BASE}/insights/trends?track=${encodeURIComponent(track)}&period=${encodeURIComponent(period)}`)
  return response.json()
}

export async function getInsightDistribution(track: string, period: string): Promise<DistributionResponse> {
  const response = await fetch(`${API_BASE}/insights/distribution?track=${encodeURIComponent(track)}&period=${encodeURIComponent(period)}`)
  return response.json()
}

export async function getInsightComparison(): Promise<ComparisonResponse> {
  const response = await fetch(`${API_BASE}/insights/comparison`)
  return response.json()
}

export async function getInsightActivities(track: string, limit: number = 5): Promise<ActivitiesResponse> {
  const response = await fetch(`${API_BASE}/insights/activities?track=${encodeURIComponent(track)}&limit=${limit}`)
  return response.json()
}
